### PR TITLE
Business Account Endpoint

### PR DESCRIPTION
**Pull request by juliamarco**
### This Pull Request Implements:

- Build Endpoint /api/v1/businesses/account for getting the account info for a business
- Add tests for happy and sad path 
- If the request is successful, it will return back the business, and it's distributor and representative and information, as well as a 200 status
- If the business could not be found in our DB, it will return a 404
- [x] Linked To User Story on Trello

### CODE REVIEW

**Comment by juliamarco** [comment]
[Business account details endpoint](https://trello.com/c/z2rv01n5/10-business-account-details-endpoint)

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -18,4 +18,8 @@ def do_authentication(user, password)
       render json: { unauth: 'Incorrect Password' }, status: 401
     end
   end
+
+  def get_business(api_key)
+    @business ||= Business.find_by!(api_key: api_key)
```
Comment (Mackenzie-Frey) [inline app/controllers/application_controller.rb]: I like that you memoized this!

File: app/models/business.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ class Business < ApplicationRecord
   has_secure_password
 
   belongs_to :distributor
+  belongs_to :representative
```
Comment (Mackenzie-Frey) [inline app/models/business.rb]: Thanks for creating this relationship and the migration.

File: spec/requests/api/v1/business_registration_spec.rb [hunk 3]
```diff
@@ -39,19 +42,21 @@
 
       expect(response).to_not be_successful
       message = JSON.parse(response.body)
-      expect(message).to eq({"name"=>["can't be blank"]})
+      expect(message["name"]).to eq(["can't be blank"])
+      expect(message["representative"]).to eq(["must exist"])
```
Comment (Mackenzie-Frey) [inline spec/requests/api/v1/business_registration_spec.rb]: I'll update swagger to reflect this.

File: spec/requests/api/v1/items_request_spec.rb [hunk 4]
```diff
@@ -16,8 +16,10 @@
                                           password: 'other'
                                           )
 
-      business_1 = create(:business, api_key: '111111111', distributor: distributor_1)
-      create(:business, api_key: '999999', distributor: distributor_2)
+      rep = create(:representative, distributor: distributor_1)
```
Comment (Mackenzie-Frey) [inline spec/requests/api/v1/items_request_spec.rb]: Thanks for updating the other tests.

**Comment by Mackenzie-Frey** [comment]
Error is restricted to my machine and occurring on other branches. Going to go ahead and merge per team Slack discussion.

**Pull request merged**